Core/src/org/sleuthkit/autopsy/healthmonitor/ServicesHealthMonitor.java

```diff
@@ -0,0 +1,767 @@
+
+        String hostName;
+        try {
+            hostName = java.net.InetAddress.getLocalHost().getHostName();
+        } catch (java.net.UnknownHostException ex) {
+            // Write it to the database but log a warning
+            logger.log(Level.WARNING, "Unable to look up host name");
+            hostName = "unknown";
+        }
+        
+        // Check if there's anything to report (right now we only have the timing map)
+        if(timingMapCopy.keySet().isEmpty()) {
+            return;
+        }
+        
+        // Write to the database
+        CoordinationService.Lock lock = getSharedDbLock();
+        if(lock == null) {
+            throw new HealthMonitorException("Error getting database lock");
+        }
+        
+        try {
+            Connection conn = connect();
+            if(conn == null) {
+                throw new HealthMonitorException("Error getting database connection");
+            }
+
+            // Add timing metrics to the database
+            String addTimingInfoSql = "INSERT INTO timing_data (name, host, timestamp, count, average, max, min) VALUES (?, ?, ?, ?, ?, ?, ?)";
+            try (PreparedStatement statement = conn.prepareStatement(addTimingInfoSql)) {
+
+                for(String name:timingMapCopy.keySet()) {
+                    TimingInfo info = timingMapCopy.get(name);
+
+                    statement.setString(1, name);
+                    statement.setString(2, hostName);
+                    statement.setLong(3, System.currentTimeMillis());
+                    statement.setLong(4, info.getCount());
+                    statement.setLong(5, info.getAverage());
+                    statement.setLong(6, info.getMax());
+                    statement.setLong(7, info.getMin());
+
+                    statement.execute();
+                }
+
+            } catch (SQLException ex) {
+                throw new HealthMonitorException("Error saving metric data to database", ex);
+            } finally {
+                try {
+                    conn.close();
+                } catch (SQLException ex) {
+                    logger.log(Level.SEVERE, "Error closing Connection.", ex);
+                }
+            }
+        } finally {
+            try {
+                lock.release();
+            } catch (CoordinationService.CoordinationServiceException ex) {
+                throw new HealthMonitorException("Error releasing database lock", ex);
+            }
+        }
+    }
+    
+    /**
+     * Check whether the health monitor database exists.
+     * Does not check the schema.
+     * @return true if the database exists, false otherwise
+     * @throws HealthMonitorException 
+     */
+    private boolean databaseExists() throws HealthMonitorException {       
+        try {
+            // Use the same database settings as the case
+            CaseDbConnectionInfo db = UserPreferences.getDatabaseConnectionInfo();
```

Comment:
Can the connection pool be used here?

Reply:
I don't think so? The connection pool connects to the actual database so I don't think I can use it to create the database or cleanly test whether it exists.

Reply:
Ok, I get it now.